packages/test/test-service-load/src/nodeStressTest.ts

```diff
@@ -138,13 +166,93 @@ async function orchestratorProcess(
                     env: envVar,
                 },
             );
+
+            if (args.enableMetrics === true) {
+                setupTelemetry(runnerProcess, logger, index, username);
+            }
+
             return new Promise((resolve) => runnerProcess.once("close", resolve));
         }));
-    } finally{
+    } finally {
+        if (logger !== undefined) {
+            await logger.flush();
+        }
         await safeExit(0, url);
     }
 }
 
+/**
+ * Setup event and metrics telemetry to be sent to loggers.
+ */
+function setupTelemetry(
+    process: child_process.ChildProcess,
+    logger: ITelemetryLogger,
+    runId: number,
+    username?: string) {
+    logger.send({
+        category: "metric",
+        eventName: "Runner Started",
+        value: 1,
+        username,
+        runId,
+    });
+
+    process.once("error", (e) => {
+        logger.send({
+            category: "metric",
```

Comment:
I assume we can't directly call some API (and instead use telemetry to communicate data) due to multi-process nature test harness, right?

Reply:
We can call APIs directly, but this way its decoupled as rest of logging where appinsights logger can be plugged in. This also gives flexibility to avoid static dependency appinsights lib on runner, and we can plugin other type of telemetry systems.